internal/gengapic/gengapic.go

```diff
@@ -365,15 +365,17 @@ func (g *generator) initializeAutoPopulatedFields(servName string, m *descriptor
 	g.imports[pbinfo.ImportSpec{Path: "github.com/google/uuid"}] = true
 	p := g.printf
 	for _, apf := range apfs {
-		f := snakeToCamel(apf.GetName())
+		getF := buildAccessor(apf.GetName(), false)
+		f := buildAccessor(apf.GetName(), true)
 		if apf.GetProto3Optional() {
 			// Type will be *string if field has explicit presence.
-			p("if req != nil && req.%s == nil {", f)
+			p("if req != nil && req%s == nil {", getF)
```

Comment:
Again, I should've been more clear, I'm sorry. With `Proto3Optional` fields, the only way to check for presence is to use direct access e.g. `req.Foo == nil`. Using the accessor will return the default empty value e.g. `""` but won't convey if it was explicitly set to that.

So in the proto3optional branch, we need to check presence with direct access. But with non-proto3optional, we want to use the accessor.

Reply:
Thanks that's clear now, updated!